### PR TITLE
partners.google-vertexai: Refactor MatchingEngine

### PR DESCRIPTION
**Pull request by jzaldi**
**Description**:

1. Break down `VertexAIVectorStore` in three components:

- `Searcher` Stores the index, embeddings and retrieves the nearest neighbors. With implementations for Public endpoint and VPC endpoint

- `DocumentStorage` Stores the documents. Implementations for Google Cloud Storage and Data Store

- `Embeddings` Langchain embeddings class.

2. Added integration tests

### CODE REVIEW

**Review by baskaryan — COMMENTED** [review]
cc @lkuligin

File: libs/partners/google-vertexai/langchain_google_vertexai/vectorstores.py [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+        self.document_id_key = document_id_key
+
+    @property
+    def embeddings(self) -> Embeddings:
+        return self.embedding
+
+    def _validate_google_libraries_installation(self) -> None:
+        """Validates that Google libraries that are needed are installed."""
+        try:
+            from google.cloud import aiplatform, storage  # noqa: F401
```
Comment (lkuligin) [inline libs/partners/google-vertexai/langchain_google_vertexai/vectorstores.py:105]: we can move it on top of the file now, `aiplatform` is a requirement for the package

File: libs/partners/google-vertexai/langchain_google_vertexai/vectorstores.py [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+            from google.oauth2 import service_account  # noqa: F401
+        except ImportError:
+            raise ImportError(
```
Comment (lkuligin) [inline libs/partners/google-vertexai/langchain_google_vertexai/vectorstores.py:108]: not needed anymore

File: libs/partners/google-vertexai/langchain_google_vertexai/vectorstores.py [hunk 3]
```diff
@@ -0,0 +1,686 @@
...
+        """
+
+        return storage.Client(
+            credentials=credentials,
+            project=project_id,
+            client_info=get_client_info(module="vertex-ai-matching-engine"),
+        )
+
+    @classmethod
+    def _init_aiplatform(
```
Comment (lkuligin) [inline libs/partners/google-vertexai/langchain_google_vertexai/vectorstores.py:545]: nits: do we really need this as a separate `classmethod`?

**Comment by jzaldi** [comment]
Closing because is going to be merged in Google's repo

**Pull request closed without merging**